src/amo/components/AddonMoreInfo/index.js

```diff
@@ -54,6 +78,38 @@ export class AddonMoreInfoBase extends React.Component<InternalProps> {
       });
     }
 
+    const addonType = addon.type;
```

Comment:
I suggest that all of the logic below could be moved into the `categories` reducer. There would be a function called, for example, `getCategoryNames`, which could be passed an array of category `slug`s and would return an array of category names. This function could be called in `mapStateToProps`.